### PR TITLE
Rube-Goldberg Class Engineering Project (Gone OSHA)

### PR DESCRIPTION
**Pull request by Latentish**
## About The Pull Request
rube

Welcome to the Rube-Goldberg Class PowerPlant! The only testing ship designed by a fuckup engineering grad student who built it with their bare hands! Learn to set up a TEG with the RUBE! Holodisks explaining a specialized and easy to use/learn atmos system! Hobbled together spaces cramped in size! Enjoy your hell of a time on this floating OSHA violation with your buddy the Chief at Engineering!

## Why It's Good For The Game
Currently there are no ships that are designed completely around giving engineers toys to play with. This ship comes with a garage full of equipment to make most of the content available to them. This ship is easy to learn, especially if you're unfamiliar with atmos or the TEG. This comes with a set of 6 Holodisks on the operation and special mechanics of this ship, and a special 7th one hidden somewhere! Wow your players with the new RUBE. And give me cookies because it's a good addition to our roster.

## Changelog
:cl:
add: Added the Rube-Goldberg Class Engineering Project ship.
imageadd: Added a steering wheel (sprite by Dexxiol)
/:cl:

### CODE REVIEW

**Comment by Latentish** [comment]
Might expand the engine room so that there's more room for construction projects.

**Comment by Latentish** [comment]
While I don't intend to make it in this PR, I would really enjoy a way to transfer power from one ship to another. Even if it was just SMES that can be moved around while still holding power. Preferably it'd be more like batteries you fill up with power before hooking it up to a SMES. Or even a wireless power transfer device you use by touching a relay to another ship's power grid. Or a power relay system that works when you aren't docked with the power providing ship, who sells it to you for research points, credits, cargo points, vendor points, or custom store. (I know that last one is out there, sue me), but any of these would give this ship something more to do with all that power they'll be making.
Without this existing my current idea is to give it some circuit kits for higher power consumption products. Maybe something to promote planet colonization type construction projects. Because as is, it's kind of a power plant for the sake of Engineering and Atmos Tech gameplay. Which I think is fine on its own, but for players like me who've done that quite a lot already, I'd like more goals. And I think that colony building goals would suit this ship well.

**Comment by MusselMain** [comment]
Gud damn this thing looks like what happens when engineers don't follow the instructions, I love it

**Comment by MusselMain** [comment]
> While I don't intend to make it in this PR, I would really enjoy a way to transfer power from one ship to another. Even if it was just SMES that can be moved around while still holding power. Preferably it'd be more like batteries you fill up with power before hooking it up to a SMES. Or even a wireless power transfer device you use by touching a relay to another ship's power grid. Or a power relay system that works when you aren't docked with the power providing ship, who sells it to you for research points, credits, cargo points, vendor points, or custom store. (I know that last one is out there, sue me), but any of these would give this ship something more to do with all that power they'll be making. Without this existing my current idea is to give it some circuit kits for higher power consumption products. Maybe something to promote planet colonization type construction projects. Because as is, it's kind of a power plant for the sake of Engineering and Atmos Tech gameplay. Which I think is fine on its own, but for players like me who've done that quite a lot already, I'd like more goals. And I think that colony building goals would suit this ship well.

Isn't the Bubble designed to be a colony ship? I mostly agree, sometimes new ships should help create different playstyles, however I also feel that your argument can be used against ships like the pill series, which technically all follow the same main idea of hardcore mode, but for those I think everyone gets a bit of fun out of them, so it doesn't matter as much, personally I feel that not every addition should be ground shatteringly unique, but I do get where you're coming from

**Comment by Latentish** [comment]
> Isn't the Bubble designed to be a colony ship? I mostly agree, sometimes new ships should help create different playstyles, however I also feel that your argument can be used against ships like the pill series, which technically all follow the same main idea of hardcore mode, but for those I think everyone gets a bit of fun out of them, so it doesn't matter as much, personally I feel that not every addition should be ground shatteringly unique, but I do get where you're coming from

Are you Referring to the idea of adding colony elements, or adding a way to mass store and move electricity? When it comes to the colony element, the idea was that an engineer crew could go to a planet, build a shack, and use the ship as a power factory. But reinforcing that idea ends at giving them a note or wood.

**Comment by Latentish** [comment]
I'll fix those stray conveyor belts later

**Comment by Apogee-dev** [comment]
what's the lore for this monstrosity anyway? it really smacks of a particularly anarchistic GEC ship to me, and you might want to theme the outfits and job titles to match. The only ship that uses those and their uniforms right now is the kugelblitz anyway.

**Comment by Latentish** [comment]
> 

The idea I had originally was making a ship that was comprised of a bunch of small cramped compartments welded together. I eventually blocked out shapes that worked for what I wanted and it took on this sort of "Engineering Graduate gets a grant to make their own ship by hand" sort of design. They blew most of the money on equipment, and just shoved it in storage or wherever they could fit it. And they spent the budget to cover everything they could need, at the cost of a single solid hull. The job names are similarly themed as if they were named by this same student. "Chief at Engineering", "Chief at Boozineering", "Engineering/Atmospherics/Mining Specialist". Names that get the point across aren't the common vernacular, but aren't very inspired.
It's supposed to feel like a huge ship you're on, but with only cramped spaces to move around in.

**Comment by Latentish** [comment]
image
image
Hm

**Comment by Latentish** [comment]
First day of TM got a lot of good feedback. Nothing in the ship is insanely broken or unreasonable. So it should be good to stay TM'd to get more feedback. Do not merge this until I get a patch up.

**Comment by Latentish** [comment]
Last patch also added the missing Particle Accelerator part, and fixed the suit storage unit that wasn't accepting items.

**Comment by Vexylius** [comment]
this ship is scary
don't feed it to your children

#INFkW/delamination

**Comment by Latentish** [comment]
> this ship is scary don't feed it to your children
> 
> #INFkW/delamination

Almost a haiku

**Comment by Dexxiol** [comment]
what the actual fuck is this ship jesus christ

File: _maps/configs/engineeringmess.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"namelists": ["GENERAL", "NATURAL", "ENGINEERING"],
+	"map_name": "Rube Goldberg-class Engineering Project",
+	"map_short_name": "Rube Goldberg-class",
+	"map_path": "_maps/shuttles/shiptest/engineeringmess.dmm",
+	"map_id": "engineeringmess",
+	"limit": 1,
+	"job_slots": {
+		"Chief at Engineering": {
+			"outfit": "/datum/outfit/job/ce",
+			"slots": 1
```
Comment (Apogee-dev) [inline _maps/configs/engineeringmess.json:14]: "outfit": "/datum/outfit/job/ce",
"officer": true,
"slots": 1

there's a flag for officer positions. the first position serves as captain but adding an officer flag gives him the officer symbol on the manifest.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

File: _maps/configs/engineeringmess.json (hunk 1 above)
Comment (Latentish) [reply _maps/configs/engineeringmess.json:14]: Good catch.

**Pull request merged**